src/services/mongoose/wallets.ts

```diff
@@ -17,7 +17,8 @@ export interface WalletRecord {
   _accountId: Types.ObjectId
   type: string
   currency: string
-  onchain: OnChainMongooseType[]
+  // TODO: even though mongoose should set a default this value has been observed to be null in prod
+  onchain: OnChainMongooseType[] | null
```

Comment:
use undefined?

Reply:
I have updated the type to reflect reality in production - there it is `null`.